feat: improve performance for multithreaded uploading

arvgord opened the pull request:
- Improved performance for multithreaded uploading

For better performance user should upload files to a DICOM Store folder, not a Series folder.
Also for better performance uploaded files should be with an extension(.dcm - etc).

dzelemba commented:
This is a very large PR... can it be split up into smaller ones? E.g. split out supporting navigation vs. improved multithreading performance. Or maybe do the refactor first in a separate PR and then follow up with these.

arvgord commented:
Ok, I'll split.

arvgord commented:
> This is a very large PR... can it be split up into smaller ones? E.g. split out supporting navigation vs. improved multithreading performance. Or maybe do the refactor first in a separate PR and then follow up with these.

Done. 
I renamed DicomFuseHelperTest to DicomFuseHelperIT because this test is more of the integration test than the unit test. In addition, I refactored the DicomFuseHelperIT to get a more realistic test.

dzelemba commented on src/test/java/com/google/dicomwebfuse/fuse/DicomFuseHelperIT.java:
```diff
@@ -0,0 +1,213 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class DicomFuseHelperIT {
```
Let's keep this named DicomFuseHelperTest... anything that doesn't require any external resources I consider a unit test.

arvgord replied:
Done

arvgord commented:
> Can you explain what changes are being made and why in the commit description?

Done.

> A summary of how much this improves performance would also be useful.

I've presented short info in the commit. I'll provide detailed performance info on the Wiki page.

arvgord commented:
@dzelemba Could you review, please?
